sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/catalog/interface.scala

```diff
@@ -51,6 +51,42 @@ import org.apache.spark.sql.types._
 import org.apache.spark.sql.util.{CaseInsensitiveStringMap, SchemaUtils}
 import org.apache.spark.util.ArrayImplicits._
 
+/**
+ * Interface providing util to convert JValue to String representation of catalog entities.
+ */
+trait MetadataHashMapTranslator {
```

Comment:
After reading the updated code, I have a new idea
```
trait WithMetadataMap {
  def toJsonLinkedHashMap: mutable.LinkedHashMap[String, JValue]  // abstract method

  def toLinkedHashMap: mutable.LinkedHashMap[String, String] = {
    jsonToString(toJsonLinkedHashMap)
  }

  protected def jsonToString ...
}
```

Then the subclasses do not need to implement `toLinkedHashMap`